### PR TITLE
Protocol limit

### PR DESCRIPTION
**Pull request by janLo**
This allowa to limit the available protocols to have less noise and better performance.

### CODE REVIEW

**Comment by janLo** [comment]
Thank you for your feedback. I'll try to clean up the commits on a best effort basis. The api changes are mainly fixes for wrong api-usage of the arduino- and pilight api.

File: src/ESPiLight.cpp [hunk 1]
```diff
@@ -18,37 +18,36 @@
...
+uint8_t ESPiLight::minrawlen = 20;
+uint8_t ESPiLight::maxrawlen = MAXPULSESTREAMLENGTH;
+uint16_t ESPiLight::mingaplen = 5100;
+uint16_t ESPiLight::maxgaplen = 10000;
 
 static void fire_callback(protocol_t *protocol, ESPiLightCallBack callback);
 
 void ESPiLight::initReceiver(byte inputPin) {
-  int interrupt = digitalPinToInterrupt(inputPin);
+  auto interrupt = (uint8_t) digitalPinToInterrupt(inputPin);
```
[the remark this answers is not in the document]
Comment (janLo) [reply src/ESPiLight.cpp]: Oh, this is my fault. `interrupt` need to be `uint8_t` but `digitalPinToInterrupt()` can actually give -1 back. So we need to cast this later.

File: src/ESPiLight.cpp [hunk 2]
```diff
@@ -295,7 +310,7 @@ int ESPiLight::parsePulseTrain(uint16_t *pulses, int length) {
         }
 
         /* Reset # of repeats after a certain delay */
-        if (((int) protocol->second - (int) protocol->first) > 500000) {
+        if (((int) protocol->second - (int) protocol->first) > 300000) {
```
[the remark this answers is not in the document]
Comment (janLo) [reply src/ESPiLight.cpp]: Ah yes, I actually want to make this configurable.

**Comment by janLo** [comment]
The changes are split in seperate commits and are in the PR #15, #16, #18 and #19. This PR is therefore obsolete.

File: src/ESPiLight.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (puuu) [reply src/ESPiLight.cpp]: Make it a constant or configurable is the way to go.

**Pull request closed without merging**